### PR TITLE
Move db related basic functions to models/db

### PR DESCRIPTION
**Pull request by lunny**
Since `models` becomes more and more big, splitting it becomes a task to refactor.

This PR create a new subpackage `models/db` and move common database functions to that package.

The next PRs should create business logic related subpackages of `models` and move related codes there. i.e. Moving all `models/issue*.go` to `models/issues/`.

### CODE REVIEW

File: modules/emoji/emoji_data.go [hunk 1]
```diff
@@ -576,7 +576,7 @@ var GemojiData = Gemoji{
 	{"\U0001f39e\ufe0f", "film frames", []string{"film_strip"}, "7.0", false},
 	{"\U0001f1eb\U0001f1ee", "flag: Finland", []string{"finland"}, "6.0", false},
 	{"\U0001f525", "fire", []string{"fire"}, "6.0", false},
-	{"\U0001f692", "fire engine", []string{"fire_engine"}, "6.0", false},
+	{"\U0001f692", "fire db.Engine", []string{"fire_engine"}, "6.0", false},
```
Comment (KN4CK3R) [inline modules/emoji/emoji_data.go]: nope

File: models/list_options.go [hunk 2]
```diff
@@ -30,8 +31,8 @@ func setSessionPagination(sess *xorm.Session, p Paginator) *xorm.Session {
 	return sess.Limit(take, skip)
 }
 
-// setSessionPagination sets pagination for a database engine
-func setEnginePagination(e Engine, p Paginator) Engine {
+// setSessionPagination sets pagination for a database db.Engine
```
Comment (KN4CK3R) [inline models/list_options.go]: nope

**Review by KN4CK3R — APPROVED** [review]
Otherwise lgtm

File: models/list_options.go (hunk 2 above)
Comment (lunny) [reply models/list_options.go]: Good catch, both done.

**Review by zeripath — APPROVED** [review]

**Pull request merged**